### ISSUE TITLE
need a material 3 version of FloatingActionButtonSpeedDial

### ISSUE THREAD

**Issue by stephentalley** [leinardi/FloatingActionButtonSpeedDial#190]
### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of the library.

### Step 2: Describe your environment
 - Library version: `com.leinardi.android:speed-dial.compose:1.0.0-alpha04`
 - Android version: `any`
 - Support library version: `any`

### Step 3: Describe the problem:

FloatingActionButtonSpeedDial doesn't conform to the [newest Material 3 spec](https://m3.material.io/components/floating-action-button/overview) for Floating Action Buttons.

In particular, the new Material 3 style is to use rounded-square buttons:

![screenshot-2022 11 22 18 17 58](https://user-images.githubusercontent.com/3507182/203440912-d489cb6a-4b9a-4168-a667-45dfdd9178e3.png)

It would also be nice to rely on the androidx.compose.material3.MaterialTheme that the rest of our app uses, rather than the androidx.compose.material.MaterialTheme used currently.

**Comment by leinardi** [comment]
Hi, I actually have already done some work to support Material 3 (source available on the m3 branch):
![image](https://user-images.githubusercontent.com/273338/203525964-908e2d2d-2a32-4d41-8458-b373df314b08.png)

I didn't announced officially, but there is an early alpha release for compose that you could try: `com.leinardi.android:speed-dial.compose:2.0.0-alpha01`.

If you do, please let me know what you think.

**Comment by daniele-athome** [comment]
Hi! Any chance you could also publish an alpha release for Classic Views? Thanks!

**Comment by leinardi** [comment]
Hi @daniele-athome, right now I can't release an alpha for the classic view, but you can try the m3 branch with the work in progress using jitpack: https://jitpack.io/#leinardi/FloatingActionButtonSpeedDial/m3-SNAPSHOT

**Comment by daniele-athome** [comment]
Thanks. I tried before, but it won't build :-( https://jitpack.io/com/github/leinardi/FloatingActionButtonSpeedDial/m3-2.0.0-alpha01-g149f241-1/build.log

For the moment I've copied the project code as a module in my app. I'll follow the upgrades here and switch back to a normal dependency when it'll be ready.

**Comment by stephentalley** [comment]
> If you do, please let me know what you think.

@leinardi The m3 alpha looks really nice, thanks!

The only minor feedback I have is that it uses Icons.Default.*, which, [according to some](https://stackoverflow.com/a/65665564/920849l), should be avoided due to the size of the icon lib dependency. But maybe that's moot if your lib strips out unused resources beforehand?

One other minor point -- you may want to choose a different maven coordinate for the m3 version, rather than just reving the major version. That way you can maintain both in parallel.

Thanks again for the m3. Any idea when you'll move it to stable?

**Comment by leinardi** [comment]
Hi @stephentalley, thanks for the feedback, it's really appreciated :+1: 

> The only minor feedback I have is that it uses Icons.Default.*, which, [according to some](https://stackoverflow.com/a/65665564/920849l), should be avoided due to the size of the icon lib dependency. But maybe that's moot if your lib strips out unused resources beforehand?

Regarding the `androidx.compose.material:material-icons-extended`, the library itself does not use it, it's not even listed as dependency:
https://github.com/leinardi/FloatingActionButtonSpeedDial/blob/m3/library-view/build.gradle#L47

It's only the sample app that uses it.

> One other minor point -- you may want to choose a different maven coordinate for the m3 version, rather than just reving the major version. That way you can maintain both in parallel.

I'm not sure, I frankly don't want to keep maintaining both versions. My plan is to release M3 as a new major version and stop supporting M2, similar to what I did with the migration from support library to androidx.

> Thanks again for the m3. Any idea when you'll move it to stable?

Currently I don't have a precise idea, I'm still collecting feedback. But, if you think that the current state is good enough, I may consider releasing an alpha version soon-ish.
